dkist/dataset/tiled_dataset.py

```diff
@@ -150,6 +128,27 @@ def __init__(
         self._data = np.ma.masked_array(dataset_array, dtype=object, mask=mask)
         meta = meta or {}
         inventory = meta.get("inventory", inventory or {})
+
+        # If headers are saved as one Table for the whole TiledDataset, use those first
+        # Otherwise stack the headers saved for component Datasets
+        if meta.get("headers") is None:
+            meta["headers"] = vstack(
+                [Table(ds.headers) if ds else Table({}) for ds in self._data.compressed()]
+            )
+        # Then distribute headers (back) out to component Datasets as slices of the main Table
+        # for row in dataset_array:
+        #     for ds in row:
+
+        offset = 0
+        for ds in self._data.compressed():
+            if ds:
+                if isinstance(ds.headers, dict) and "offset" in ds.headers.keys():
```

Comment:
I don't think this should be here, any offset / shape thing should just be in the asdf / asdf converters.

Reply:
Fairly sure you said it should be in the constructor but I don't remember why :smile: